pkg/virt-launcher/virtwrap/manager_test.go

```diff
@@ -1622,6 +1624,95 @@ var _ = Describe("Manager", func() {
 			Expect(err).ToNot(HaveOccurred())
 			Expect(domSpec).ToNot(BeNil())
 		})
+
+		Context("on call to GetGuestOSInfo", func() {
+			var libvirtmanager DomainManager
+			var agentStore agentpoller.AsyncAgentStore
+
+			BeforeEach(func() {
+				agentStore = agentpoller.NewAsyncAgentStore()
+				libvirtmanager, _ = NewLibvirtDomainManager(mockConn, testVirtShareDir, nil, 0, &agentStore, "/usr/share/OVMF")
+			})
+
+			It("should return nil when no osInfo exists in the cache", func() {
+				osInfo := libvirtmanager.GetGuestOSInfo()
+				Expect(osInfo).To(BeNil())
+			})
+
+			It("should return osInfo when osInfo exists in the cache", func() {
+				fakeInfo := api.GuestOSInfo{
+					Name: "TestGuestOSName",
+				}
+
+				agentStore.Store(agentpoller.GET_OSINFO, fakeInfo)
+				osInfo := libvirtmanager.GetGuestOSInfo()
+				Expect(reflect.DeepEqual(&fakeInfo, osInfo)).To(BeTrue())
+			})
+		})
+
+		Context("on call to InterfacesStatus", func() {
+			var libvirtmanager DomainManager
+			var agentStore agentpoller.AsyncAgentStore
+			domInterfaces := []api.Interface{
```

Comment:
Please be consistent here, if you call one info "fake", do the same to all. I do not really understand why it needs to be called a fake, as it is a regular data, but I do not really mind much.